test/framework/clusterctl/client.go

```diff
@@ -103,7 +103,13 @@ func InitWithBinary(_ context.Context, binary string, input InitInput) {
 
 	out, err := cmd.CombinedOutput()
 	_ = os.WriteFile(filepath.Join(input.LogFolder, "clusterctl-init.log"), out, 0644) //nolint:gosec // this is a log file to be shared via prow artifacts
-	Expect(err).ToNot(HaveOccurred(), "failed to run clusterctl init:\n%s", string(out))
+	var stdErr string
+	if err != nil {
+		if exitErr, ok := err.(*exec.ExitError); ok {
+			stdErr = string(exitErr.Stderr)
+		}
+	}
+	Expect(err).ToNot(HaveOccurred(), "failed to run clusterctl init:\nstdout:\n%s\nstderr:\n%s", string(out), stdErr)
```

Comment:
This way we can read the stdErr and don't have to convert the byte array which is printed by printing the err